frontend/src/components/LoadMoreButton/LoadMoreButton.jsx

```diff
@@ -12,7 +12,12 @@ const LoadMoreButton = ({
   setOffset,
   setImages,
 }) => {
-  const [isLoading, setIsLoading] = useState(false);
+    const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    window.addEventListener('scroll', handleScroll);
+    return () => window.removeEventListener('scroll', handleScroll);
+  });
```

Comment:
this looks wrong to me. I dont think we should add and remove scroll event listener on every rerender.